crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_pre_authorization_status.rs

```diff
@@ -0,0 +1,392 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Poll PreAuthorization Status (Public)
+// ==================
+impl SargonOS {
+    /// Polls the state of a PreAuthorization until we can determine its `TransactionStatus`.
+    pub async fn poll_pre_authorization_status(
+        &self,
+        intent_hash: SubintentHash,
+        expiration: Option<DappToWalletInteractionSubintentExpiration>,
+    ) -> Result<TransactionStatus> {
+        let (status, _) = self
+            .poll_subintent_status_with_delays(intent_hash, expiration)
+            .await?;
+
+        Ok(status)
+    }
+}
+
+// ==================
+// Poll PreAuthorization Status (Internal)
+// ==================
+impl SargonOS {
+    /// Polls the state of a Subintent until it is submitted
+    ///
+    /// It returns the `TransactionIntentHash`, but also the list of delays between each poll.
+    async fn poll_subintent_status_with_delays(
```

Comment:
If much rather we put the logic in a shared implementation used by tx v1 and subintents.

Because this looks like big copy paste with same logic ? Or did I miss something?

Reply:
I've added a note [here](https://github.com/radixdlt/sargon/pull/247/files/bdbe33d2a28d5c2ba9e92d796ad0bec4282f428f#diff-927cc7b4a971b1591828417003462b78720089019f792be45afdcd0eadef7ce8R79) explaining that once we know for sure how the endpoint is going to work, we may want to reuse the same logic.

However, as we still don't know if it is gonna be a different endpoint/model, it doesn't make too much sense to extract the common logic yet.